File: src/SchoolLunchMenu/Models/MenuCache.cs
namespace SchoolLunchMenu.Models;

using SchoolLunchMenu.Models.Api;

/// <summary>
/// Disk-persisted cache of the last fetched menu data and allergens,
/// allowing the app to preload data on startup without an API call.
/// </summary>
public class MenuCache
{
    /// <summary>When this cache was written (UTC).</summary>
    public DateTime SavedAtUtc { get; set; }

    /// <summary>The cached menu response.</summary>
    public FamilyMenuResponse? MenuResponse { get; set; }

    /// <summary>The cached allergen list.</summary>
    public List<AllergyItem>? Allergies { get; set; }

    /// <summary>The cached identifier response (district + buildings).</summary>
    public FamilyMenuIdentifierResponse? IdentifierResponse { get; set; }
}

File: school-lunch-menu.Mac/SchoolLunchMenu/Models/MenuCache.swift
import Foundation

// MARK: - MenuCache

/// Disk-persisted cache of the last fetched menu data and allergens,
/// allowing the app to preload data on startup without an API call.
struct MenuCache: Codable {
    /// When this cache was written (UTC).
    var savedAtUtc: Date

    /// The cached menu response.
    var menuResponse: FamilyMenuResponse?

    /// The cached allergen list.
    var allergies: [AllergyItem]?

    /// The cached identifier response (district + buildings).
    var identifierResponse: FamilyMenuIdentifierResponse?

    init(
        savedAtUtc: Date = Date(),
        menuResponse: FamilyMenuResponse? = nil,
        allergies: [AllergyItem]? = nil,
        identifierResponse: FamilyMenuIdentifierResponse? = nil
    ) {
        self.savedAtUtc = savedAtUtc
        self.menuResponse = menuResponse
        self.allergies = allergies
        self.identifierResponse = identifierResponse
    }
}
